core/src/main/java/org/apache/calcite/sql/validate/SqlUserDefinedAggFunction.java

```diff
@@ -54,7 +54,8 @@ public SqlUserDefinedAggFunction(SqlIdentifier opName,
       SqlOperandTypeInference operandTypeInference,
       SqlOperandTypeChecker operandTypeChecker, AggregateFunction function,
       boolean requiresOrder, boolean requiresOver, RelDataTypeFactory typeFactory) {
-    super(Util.last(opName.names), opName, SqlKind.OTHER_FUNCTION,
+    super(Util.last(opName.names), opName, SqlKind.getSqlKindByName(opName.names.get(0)) != null
+                    ? SqlKind.valueOf(opName.names.get(0)) : SqlKind.OTHER_FUNCTION,
```

Comment:
I agree with @vlsi. It's OK to have a UDAF with the same name as a built-in, but it's not OK for it to impersonate it by taking on its SqlKind value.

What happens if you leave its SqlKind as OTHER_FUNCTION?